### PR TITLE
KAA-362  Default transport channel sends more than one CONNECT message on the authorization phase

### PR DESCRIPTION
**Pull request by andreypanasenko**
Review
@dyosick 
@yzeygerman

### CODE REVIEW

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 1]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+    //Checking receiving KAA_SYNC message
+    CHECK_SOCKET_RW(channel,true,true);
+
+    //Third sync
+    kaa_service_t services3[] = {KAA_SERVICE_LOGGING};
+    channel->sync_handler(channel->context, services3, 1);
+
+    //Check correct RD,WR operation, in this point we waiting for RD operations true
+    //and WR true, pending services and empty buffer.
+    CHECK_SOCKET_RW(channel,true,true);
```
Comment (izeigerman) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please insert spaces after commas.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 2]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+
+    //Imitate socket ready for reading, and read CONNACK message
+    error_code = kaa_tcp_channel_process_event(channel,FD_READ);
+    ASSERT_EQUAL(error_code, KAA_ERR_NONE);
+    ASSERT_EQUAL(access_point_test_info.connack_read, true);
+
+    //Check correct RD,WR operation, in this point we waiting for RD operations true
+    //and WR true, number of pending services and empty buffer.
+    //Checking receiving KAA_SYNC message
+    CHECK_SOCKET_RW(channel,true,true);
```
Comment (izeigerman) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please insert spaces after commas.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 3]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+    kaa_service_t services2[] = {KAA_SERVICE_PROFILE, KAA_SERVICE_EVENT};
+    channel->sync_handler(channel->context, services2, 2);
+    ASSERT_EQUAL(access_point_test_info.fill_connect_message, false);
+    ASSERT_EQUAL(access_point_test_info.request_connect, false);
+    //Check correct RD,WR operation, in this point we waiting for RD operations true
+    //and WR false, no pending services and empty buffer.
+    CHECK_SOCKET_RW(channel,true,false);
+
+    //Imitate socket ready for reading, and read CONNACK message
+    error_code = kaa_tcp_channel_process_event(channel,FD_READ);
```
Comment (izeigerman) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please insert spaces after commas.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 4]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+    //Imitate socket ready for writing, and writing CONNECT message
+    error_code = kaa_tcp_channel_process_event(channel,FD_WRITE);
+    ASSERT_EQUAL(error_code, KAA_ERR_NONE);
+    ASSERT_EQUAL(access_point_test_info.auth_packet_written, true);
+    ASSERT_EQUAL(access_point_test_info.fill_connect_message, false);
+    ASSERT_EQUAL(access_point_test_info.request_connect, false);
+    //Check correct RD,WR operation, in this point we waiting for RD operations true
+    //and WR false, no pending services and empty buffer.
+    CHECK_SOCKET_RW(channel,true,false);
+    KAA_LOG_INFO(logger,KAA_ERR_NONE,">>>>>>>>>>D4");
```
Comment (izeigerman) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please insert spaces after commas.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 5]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+    //CONNECT message
+    error_code = kaa_tcp_channel_process_event(channel,FD_WRITE);
+    ASSERT_EQUAL(error_code, KAA_ERR_NONE);
+    ASSERT_EQUAL(access_point_test_info.socket_connected, true);
+    ASSERT_EQUAL(access_point_test_info.socket_connected_callback, true);
+    ASSERT_EQUAL(access_point_test_info.fill_connect_message, true);
+    ASSERT_EQUAL(access_point_test_info.request_connect, true);
+
+    //Check correct RD,WR operation, in this point we waiting for RD,WR operations true
+    CHECK_SOCKET_RW(channel,true,true);
```
Comment (izeigerman) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please insert spaces after commas.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 6]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+    access_point_test_info.request_connect           = false;
+    access_point_test_info.auth_packet_written       = false;
+
+    kaa_service_t services2[] = {KAA_SERVICE_PROFILE, KAA_SERVICE_EVENT};
+    channel->sync_handler(channel->context, services2, 2);
+    ASSERT_EQUAL(access_point_test_info.fill_connect_message, false);
+    ASSERT_EQUAL(access_point_test_info.request_connect, false);
+    //Check correct RD,WR operation, in this point we waiting for RD operations true
+    //and WR false, no pending services and empty buffer.
+    CHECK_SOCKET_RW(channel,true,false);
```
Comment (izeigerman) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please insert spaces after commas.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c [hunk 7]
```diff
@@ -349,6 +349,115 @@ void test_kaa_tcp_channel_io_error_flow()
...
+    //and WR true, pending services and empty buffer.
+    CHECK_SOCKET_RW(channel,true,true);
+
+    test_sync_exchange(channel);
+
+    test_send_disconnect(channel);
+
+    channel->destroy(channel->context);
+
+    KAA_TRACE_OUT(logger);
```
Comment (ghost) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Please, move KAA_TRACE_OUT() to the end of the test.

File: client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c (hunk 4 above)
Comment (ghost) [inline client/client-multi/client-c/test/kaa_tcp_channel/test_kaa_tcp_channel_operation.c]: Is this log trace really needed?

**Comment by andreypanasenko** [comment]
Code style fixed, KAA_TRACE_OUT moved.

**Pull request merged**